docs(enterprise): fix contradictory single-compactor constraint in clustering docs (#7075)
influxdata/docs-v2: content/influxdb3/enterprise/admin/clustering.md

# Multiple modes
influxdb3 serve --mode=ingest,query

# All modes (default, for single-node Enterprise only)
influxdb3 serve --mode=all
```

Available modes:

- `all`: All capabilities enabled (single-node Enterprise deployments only)
- `ingest`: Data ingestion and line protocol parsing
- `query`: Query execution and data retrieval
- `compact`: Background compaction and optimization
- `process`: Data processing and transformations

> [!Warning]

> > Use `all` mode for **single-node** Enterprise deployments only.

Reviewer: ```suggestion
> #### Don't use all mode in a multi-node cluster
>
> Use `all` mode for **single-node** Enterprise deployments only.
```

> > Avoid using `all` mode in a multi-node cluster—some cluster features such as replication and catalog refresh aren't designed to work with `all`-mode nodes.

Reviewer: ```suggestion
> Some cluster features such as replication and catalog refresh aren't designed to work with `all`-mode nodes.
```

# Multiple modes
influxdb3 serve --mode=ingest,query

# All modes (default, for single-node Enterprise only)
influxdb3 serve --mode=all
```

Available modes:

- `all`: All capabilities enabled (single-node Enterprise deployments only)
- `ingest`: Data ingestion and line protocol parsing
- `query`: Query execution and data retrieval
- `compact`: Background compaction and optimization
- `process`: Data processing and transformations

> [!Warning]
> Use `all` mode for **single-node** Enterprise deployments only.

> > Avoid using `all` mode in a multi-node cluster—some cluster features such as replication and catalog refresh aren't designed to work with `all`-mode nodes.

Reviewer 2: s/Avoid using/Do Not use/

- `compact`: Background compaction and optimization
- `process`: Data processing and transformations

> [!Warning]
> Only **one** node per cluster can run in a mode that includes compaction (`compact` or `all`).
> Running multiple compactors causes data corruption.

> > In a cluster, assign `all` mode to at most one node, and ensure no other node uses the `compact` mode.

Reviewer 2: I'm not sure all mode is compatible with clustering at all. I think "all" nodes act alone, but i haven't tried it. 

Why would i run an "All" node with a separate "ingest" node for example. It's an odd configuration. If it is possible, i don't think i would recommended it.

Reviewer: Updated the issue and PR description with the latest advice.

- Growing number of small Parquet files
- Increasing query times due to file fragmentation

> **Solution:** Increase DataFusion threads on your single compactor node (see [Compactor node issues](#compactor-node-issues))

Reviewer 2: for parquet backend yes, which uses datafusion for compaction (pacha doesn't use DF for compaction).

> **Solution:** Increase DataFusion threads on your single compactor node (see [Compactor node issues](#compactor-node-issues))

Reviewer: > for parquet backend yes, which uses datafusion for compaction (pacha doesn't use DF for compaction).

@copilot Specify that this solution is for the Parquet-backed storage engine. The Performance Preview (with PachaTree) doesn't use DataFusion for compaction.

- Growing number of small Parquet files
- Increasing query times due to file fragmentation

> **Solution:** For nodes using the default Parquet-backed storage engine, increase DataFusion threads on your single compactor node (see [Compactor node issues](#compactor-node-issues)).

Reviewer: ```suggestion
**Solution:** For nodes using the Parquet-backed storage engine, increase DataFusion threads on your single compactor node (see [Compactor node issues](#compactor-node-issues)).
```

## Migrate to specialized nodes

### From single-node to specialized cluster

> [!Note]

> > `all` mode is intended for single-node Enterprise deployments.

Reviewer: ```suggestion
> `all` mode is only for single-node Enterprise deployments.
```

> > When scaling to a multi-node cluster, replace `all` with explicit modes and assign `compact` to exactly one node.

Reviewer: ```suggestion

When scaling a single `all` node cluster to a multi-node cluster:

- Replace the `all` node with nodes that have explicit, specialized modes
- Assign `compact` mode to exactly one node that uses the same node-id as the `all` node being replaced
```

> ```bash

Reviewer: @copilot Fix the notation/syntax in this "bash" example so that it represents real code. Observe the following for all examples scaling from `all` mode to a multi-node cluster:

> - Replace the `all` node with nodes that have explicit, specialized modes
> - Assign `compact` mode to exactly one node that uses the same node-id as the `all` node being replaced

## Migrate to specialized nodes

### From single-node to specialized cluster

> [!Note]
> `all` mode is intended for single-node Enterprise deployments.

> > When scaling to a multi-node cluster, replace `all` with explicit modes and assign `compact` to exactly one node.

Reviewer 2: it needs to be more than this; when upgrading from an All to multinode, the "new" `compact` node _must_ have the same node-id as the `all` node being replaced.
